jupyterlab_launcher/settings_handler.py

```diff
@@ -66,30 +57,55 @@ def patch(self, section_name):
         if not self.settings_dir:
             raise web.HTTPError(404, "No current settings directory")
 
-        path = os.path.join(self.schemas_dir, section_name + '.json')
-
-        if not os.path.exists(path):
-            raise web.HTTPError(404, "Schema not found for: %r" % section_name)
-
-        data = self.get_json_body()  # Will raise 400 if content is not valid JSON
+        # Will raise 400 if content is not valid JSON.
+        data = self.get_json_body()
 
         # Validate the data against the schema.
         if Validator is not None:
-            with open(path) as fid:
-                schema = json.load(fid)
-            validator = Validator(schema)
+            validator = Validator(_get_schema(self.schemas_dir, section_name))
             try:
                 validator.validate(data)
             except ValidationError as e:
                 raise web.HTTPError(400, str(e))
 
-        # Create the settings dir as needed.
-        if not os.path.exists(self.settings_dir):
-            os.makedirs(self.settings_dir)
-
-        path = os.path.join(self.settings_dir, section_name + '.json')
-
-        with open(path, 'w') as fid:
+        with open(_path(self.settings_dir, section_name, True), "w") as fid:
             json.dump(data, fid)
 
         self.set_status(204)
+
+
+def _path(root_dir, section_name, make_dirs = False):
+    """Parse the URL section name and find the local file system path."""
+
+    # Attempt to parse path, e.g. @jupyterlab/apputils-extension:themes.
+    try:
+        package_dir, plugin = section_name.split(":")
+        parent_dir = os.path.join(root_dir, package_dir)
+        path = os.path.join(parent_dir, plugin + ".json")
+    # This is deprecated and exists to support the older URL scheme.
+    except Exception as e:
+        path = os.path.join(root_dir, section_name + ".json")
+
+    if make_dirs and parent_dir and not os.path.exists(parent_dir):
```

Comment:
parent_dir would only be defined if there was a `:` in the section name (need to pre-define it outside of the try block).